src/components/Cocktail/Cocktail.js

```diff
@@ -1,42 +1,29 @@
 import './Cocktail.css';
 import { NavLink } from 'react-router-dom/cjs/react-router-dom.min';
 import { useState, useEffect } from 'react';
+import getCocktails from '../../api-calls';
+import listIngredients from '../../helper-functions';
 
 function Cocktail() {
 
     const [cocktail, setCocktail] = useState({});
-	const [randomError, setError] = useState('');
-
-	const getRandomCocktail = async () => {
-		try {
-			const response = await fetch('https://www.thecocktaildb.com/api/json/v1/1/random.php');
-			const data = await response.json();
-			setCocktail(data.drinks[0]);
-		} catch (error) {
-			setError(error.message);
-		}
+	const [error, setError] = useState('');
+
+	const getRandomCocktail = () => {
+		const promise = getCocktails('random.php');
+        promise.then(data => {
+            if (typeof data === 'string' || data instanceof String) {
```

Comment:
LOVE the idea of using typeof to see if it's an error or your response coming back - that's something I haven't gotten figured out with my own API.  Good thinking 🧠